_templates/component/new/component.ejs.t

```diff
@@ -3,7 +3,7 @@ to: src/app/<%=path%>/<%=name%>/<%=name%>.component.js
 ---
 'use strict';
 
-import template from './<%=name%>.html';
+import template from './<%=name%>.html?raw';
```

Comment:
Hi Ricardo, first of all, thank you for contributing to this project 👍

Could you please expand a little bit, on what this "query"-like syntax does exactly in this case? Or give links to resources. I can probably add them to README in some way.

Reply:
Hi @fyodorio!
Thanks so much for your answer. I was using this started to respond to a code challenge for an job application, but I'm not work with AngularJS on my daily basics. I was looking for an AngularJS + Vite solution to improve my developer experience by using the tools I use working with Vue.js.

After cloning the repository and analysing the project structure, I learn about the folder structure and organization. Reading the README.md, the project recommends add new components or services by calling `npx` command, that executes **hgen** to create the files for the component or service as defined on the templates.

When I added the new component on the `./app/components/components.js`, I was receiving a Vite.js error related to the added new component. I found this error was related about how Vite import the files. In the `./src/path/to/new/component/component_name/component_name.component.js`. After analyze the file, I found the template  definition when the component is exported, is expecting an string, but vite don't import the HTML as a string, but instead, parse the file in someway.

I discover on the Vite documentation about the `?raw` suffix here: [Importing Asset as String](https://vitejs.dev/guide/assets.html#importing-asset-as-string). By using this suffix on the file importation, you avoid any kind of modification when Vite process the importation.

I'm using Node v.14.18.3 to build and execute the dev server.

Reply:
OK, tested it and it works like a charm 👍 thank you man, it really helped, also I have found some stuff that needs updating in terms of components generation, so thanks again for taking time to make this PR 

![](https://media3.giphy.com/media/7aBE32jCr6lOhtuE9v/giphy.gif)

Reply:
Thanks! I'm glad to collaborate on something you build for us. Thank you.